src/uint/mul.rs

```diff
@@ -17,42 +17,22 @@ impl<const LIMBS: usize> UInt<LIMBS> {
     ///
     /// For more info see: <https://github.com/RustCrypto/crypto-bigint/issues/4>
     // TODO(tarcieri): use `concat` (or similar) when const trait is stable
-    pub const fn mul_wide(&self, rhs: &Self) -> (Self, Self) {
-        let mut i = 0;
-        let mut lo = Self::ZERO;
-        let mut hi = Self::ZERO;
-
-        // Schoolbook multiplication.
-        // TODO(tarcieri): use Karatsuba for better performance?
-        while i < LIMBS {
-            let mut j = 0;
-            let mut carry = Limb::ZERO;
-
-            while j < LIMBS {
-                let k = i + j;
-
-                if k >= LIMBS {
-                    let (n, c) = hi.limbs[k - LIMBS].mac(self.limbs[i], rhs.limbs[j], carry);
-                    hi.limbs[k - LIMBS] = n;
-                    carry = c;
-                } else {
-                    let (n, c) = lo.limbs[k].mac(self.limbs[i], rhs.limbs[j], carry);
-                    lo.limbs[k] = n;
-                    carry = c;
-                }
-
-                j += 1;
-            }
-
-            hi.limbs[i + j - LIMBS] = carry;
-            i += 1;
-        }
-
-        (lo, hi)
+    pub fn mul_wide(&self, rhs: &Self) -> (Self, Self) {
```

Comment:
Please keep this function `const fn`.

You'll need to use loops instead of iterators, but otherwise it shouldn't be too difficult.

Reply:
Yes I switched to using while loops, but we can't use &mut T in const contexts and split_at, split_at_mut are not const fns. I'll have to redesign the impl if mul_wide is to be a const fn.

Reply:
One way to move this forward would be to keep the current schoolbook multiplication method for `const fn mul_wide`, and then use the fast implementation in the `Mul` traits.

Reply:
Apparently there's a `konst` crate which implements [`split_at_mut` as a const fn](https://docs.rs/konst/latest/konst/slice/fn.split_at_mut.html).
IIUC, for it to be constant-time it needs rust version 1.64 at minimum, and it's using some unsafe code.
However, it seems like this crate's MSRV is 1.65, so perhaps that'd work?

Reply:
Would prefer not to add additional dependencies. Curious how it works though... I thought all mut references were still unstable in `const fn`

Reply:
Regarding `&mut T` in `const fn`s it seems like this is blocking that: https://github.com/rust-lang/rust/issues/57349, although I may be wrong, I feel out of my depth here.

Reply:
Yeah, I'm confused how the implementation in the `konst` crate could work on stable. It uses `from_raw_parts_mut` which is marked const: unstable, and the function doesn't compile on 1.70.0: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=79b84c558fb0ada4fa9e6980a797aa93

```
error[[E0658]](https://doc.rust-lang.org/stable/error_codes/E0658.html): mutable references are not allowed in constant functions
```